packages/commonwealth/client/scripts/views/components/user_survey_popup.tsx

```diff
@@ -0,0 +1,156 @@
+            <CWButton
+              buttonType="secondary-black"
+              label="No Thanks"
+              onclick={onClose}
+            />
+            <CWButton
+              buttonType="primary-black"
+              label="Sure Thing!"
+              onclick={onRedirectClick}
+            />
+          </div>
+          <CWCheckbox
+            checked={checked}
+            label="Please don't show this again"
+            onchange={onCheckboxClick}
+            className="checkbox"
+          />
+        </div>
+      </CWGrowl>
+    );
+  }
+}
+
+function surveyDisplayTimeElapsed() {
+  const lastSurvey = localStorage.getItem('user-survey-last-displayed');
+  if (!lastSurvey) {
+    // They have never seen the survey growl before
+    return true;
+  }
+  const lastSurveyDate = new Date(parseInt(lastSurvey, 10));
+  const now = new Date();
+
+  const timeSinceLastSurvey = now.getTime() - lastSurveyDate.getTime();
+  return timeSinceLastSurvey > USER_SURVEY_DISPLAY_INTERVAL;
+}
+
+const openTypeform = (
+  typeformBaseUrl: string,
+  params: { [param: string]: string }
+) => {
+  let paramsString = '';
+  Object.keys(params).forEach((key, i) => {
+    if (i > 0) {
+      paramsString += '&';
+    } else {
+      paramsString += '#';
+    }
+    paramsString += `${key}=${params[key]}`;
+  });
+
+  window.open(`${typeformBaseUrl}${paramsString}`, '_blank');
+};
+
+type UserSurveyPopupAttrs = {
+  surveyReadyForDisplay: boolean;
+};
+
+export class UserSurveyPopup implements m.ClassComponent<UserSurveyPopupAttrs> {
+  private surveyLocked: boolean;
+  private hideForeverChecked: boolean; // radio button indicating whether the user wants to hide the survey forever
+
+  oncreate() {
+    this.hideForeverChecked = false;
+    const surveyCurrentlyLocked = localStorage.getItem('user-survey-locked');
+    const surveyDelayTimeElapsed = surveyDisplayTimeElapsed();
+
+    if (surveyCurrentlyLocked) {
+      this.surveyLocked = true;
+    } else {
+      this.surveyLocked = !surveyDelayTimeElapsed;
+    }
+  }
+
+  view(vnode) {
+    const { surveyReadyForDisplay } = vnode.attrs;
+
+    const handleClose = () => {
+      if (this.hideForeverChecked) {
+        localStorage.setItem('user-survey-locked', 'true');
+      }
+      this.surveyLocked = true;
+      localStorage.setItem('user-survey-last-displayed', Date.now().toString());
+      console.log('setting new survey-last-displayed');
+      m.redraw();
+    };
+
+    const handleRedirect = () => {
+      const address =
+        app.user.activeAccount?.address ?? app.user.addresses[0].address;
+      const name = app.user.activeAccount?.profile.name ?? 'there';
+
+      openTypeform('https://hicommonwealth.typeform.com/to/dS5q7cM2', {
+        address,
+        name,
+      });
+
+      // We don't need to display it to this user again
+      this.surveyLocked = true;
+      localStorage.setItem('user-survey-locked', 'true');
+    };
+
+    return (
+      <UserSurveyView
```

Comment:
Why is all this logic about the Survey in this component and not generalized in the CWGrowl component? I'm not sure I understand the simplicity of the CWGrowl as a wrapper for CWCard if the component is then mostly fleshed out here? Seems like we're gonna wanna do this more than once and have cards in the future? idk let's chat about it

Reply:
This was intentional, though you may disagree with the logic and I'm happy to change. I felt the show/hide logic here was bespoke enough that it made sense to have a special Survey component that handles the particulars of this growl usage. That way the hierarchy is:

Component handling specific logic of growl
    Growl component that mounts to DOM without obstructing other elements
         Children of growl aka view component for specific growl

Reply:
Aden told me this was a one off so didn't have any other clear examples of how something similar would be used down the road and didn't want to lock anything in to the Growl component itself

Reply:
Growls are just popovers that float somewhere on the screen. 

Do they all have local-storage state about closing them _forever_? or just this survey instance? 
We use local-storage enough to probably have that as a util used around for various purposes.